### PR TITLE
Add Works with Badge to sidebar + set ESPHome as works with Local

### PR DESCRIPTION
**Pull request by zsarnett**
## Proposed change

image

## Type of change

- [ ] Spelling, grammar or other readability improvements (`current` branch).
- [x] Adjusted missing or incorrect information in the current documentation (`current` branch).
- [ ] Added documentation for a new integration I'm adding to Home Assistant (`next` branch).
  - [ ] I've opened up a PR to add logo's and icons in [Brands repository](https://github.com/home-assistant/brands).
- [ ] Added documentation for a new feature I'm adding to Home Assistant (`next` branch).
- [ ] Removed stale or deprecated documentation.

## Additional information

- Link to parent pull request in the codebase: 
- Link to parent pull request in the Brands repository: 
- This PR fixes or closes issue: fixes #

## Checklist

- [x] This PR uses the correct branch, based on one of the following:
  - I made a change to the existing documentation and used the `current` branch.
  - I made a change that is related to an upcoming version of Home Assistant and used the `next` branch.
- [x] The documentation follows the Home Assistant documentation [standards].

[standards]: https://developers.home-assistant.io/docs/documenting/standards

### CODE REVIEW

File: sass/custom/_paulus.scss [hunk 1]
```diff
@@ -831,4 +831,8 @@ a.my {
   position: relative;
   top: -100px; 
   visibility: hidden;
+}
+
+.works-with-badge {
```
Comment (balloob) [inline sass/custom/_paulus.scss]: Margins are specific to where it's used. We can't put a global class here.

File: source/_includes/asides/component_navigation.html [hunk 2]
```diff
@@ -37,6 +37,14 @@
     {%- endif -%}.
   </div>
 
+  {% if page.works_with %}
+  <div>
+    {%- for type in page.works_with -%}
+      <img src="../../images/works_with/works-with-{{ type }}.png" alt="Works via {{ type }}" class="works-with-badge">
```
Comment (balloob) [inline source/_includes/asides/component_navigation.html]: Use one of the margin helpers to add marginor wrap it in section div

Comment (balloob) [reply source/_includes/asides/component_navigation.html]: Alt text should include full text of the badge.

Comment (balloob) [reply source/_includes/asides/component_navigation.html]: Clicking on it should link to a page where a user can learn more what this means (the intro blog post)

**Review by balloob — APPROVED** [review]

File: source/_integrations/esphome.markdown [hunk 3]
```diff
@@ -29,6 +29,8 @@ ha_platforms:
   - switch
 ha_integration_type: integration
 ha_dhcp: true
+works_with: 
+  - local
```
Comment (frenck) [inline source/_integrations/esphome.markdown:33]: According to our Partner program, ESPHome doesn't live up to be allowed into this?

> Integration Quality Scale requirement of Gold or better

ESPHome doesn't meet this requirement.

File: source/_includes/asides/component_navigation.html [hunk 4]
```diff
@@ -37,6 +37,16 @@
     {%- endif -%}.
   </div>
 
+  {% if page.works_with %}
+    {%- for type in page.works_with -%}
```
Comment (frenck) [inline source/_includes/asides/component_navigation.html:41]: All custom variables are prefixed `ha_` could you please adjust it for this one to @zsarnett? Thanks 👍

File: source/_includes/asides/component_navigation.html [hunk 5]
```diff
@@ -37,6 +37,16 @@
     {%- endif -%}.
   </div>
 
+  {% if page.works_with %}
```
Comment (frenck) [inline source/_includes/asides/component_navigation.html:40]: Let's adjust this check to ensure it has a codeowner set and meets the integration quality scale requirement.

**Pull request merged**